### DISCUSSION TITLE
Allow setting Client.head_html and Client.body_html

### DISCUSSION THREAD

**Discussion by DaelonSuzuka** [zauberzeug/nicegui#2126] [Ideas / Feature Requests]
I want to insert some raw html content (style tags or including a css, but this would be useful for other things too) into the page template so it's automatically in every page that gets generated.

The Client class from client.py builds the `index.html` template, and it has two attributes, `head_html` and `body_html`, that are filled into the template.

If those two attributes were moved to the class, I think I'd be able to do this:

```py
Client.head_html = """
<style>style content blah blah</style>
<link href="my_styles.css" rel="stylesheet">
"""
```

Is there another method to do this or some downside I'm missing?

**Comment by falkoschindler** [comment]
We usually simply call a function at the beginning of each page definition to define things like head HTML:
https://github.com/zauberzeug/nicegui/blob/8c24cc82fb527b2dd09c0225c3f0224f5751a921/website/main_page.py#L12

But your PR #2127 looks very interesting. 🙂

**Reply by DaelonSuzuka** [reply]
Yeah I figured that's how people were doing it, but I just can't bring myself to add a line of code to *every route* (plus imports!) when all I want to add is `a { color: #07a !important }`. (The default link color fails Google Lighthouse checks for contrast and readability.)

I've also been looking at application telemetry/analytics lately. [Plausible](https://plausible.io/) checks a lot of boxes for me, and you install it by inserting this block in your pages `<head>`.
```html
<script defer data-domain="yourdomain.com" src="https://plausible.io/js/script.js"></script>
```

I'm sure there's plenty of other useful things somebody might want on *every* page.

**Reply by falkoschindler** [reply]
PR #2127 has been merged. ✔️
